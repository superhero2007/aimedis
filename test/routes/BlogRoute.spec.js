import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import BlogRoute from 'routes/BlogRoute';

describe('routes/BlogRoute', () => {
  it('should render the Blog Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/blog']}>
        <BlogRoute
          exact
          path="/blog"
          component={() => (<div>Blog</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the Blog Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/blog']}>
        <BlogRoute
          exact
          path="/blog"
          component={() => (<div>Blog</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
