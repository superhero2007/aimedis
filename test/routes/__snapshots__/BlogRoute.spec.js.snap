// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/BlogRoute should render the Blog Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">Blog</div>"`;

exports[`routes/BlogRoute should render the Blog Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">Blog</div>"`;
